packages/p2p/src/components/my-profile/payment-methods/add-payment-method/add-payment-method-form.jsx

```diff
@@ -2,24 +2,23 @@ import React from 'react';
 import classNames from 'classnames';
 import PropTypes from 'prop-types';
 import { observer } from 'mobx-react-lite';
-import { Field, Form, Formik } from 'formik';
+import { Field, Form } from 'formik';
 import { Button, Icon, Input, Loading, Modal, Text } from '@deriv/components';
 import { Localize, localize } from 'Components/i18next';
 import { useStores } from 'Stores';
+import ModalForm from 'Components/modal-manager/modal-form.jsx';
+import { useModalManagerContext } from 'Components/modal-manager/modal-manager-context';
 
-const AddPaymentMethodForm = ({ formik_ref, should_show_separated_footer = false }) => {
+const AddPaymentMethodForm = ({ should_show_separated_footer = false }) => {
     const { my_ads_store, my_profile_store } = useStores();
+    const { showModal } = useModalManagerContext();
 
     React.useEffect(() => {
         my_profile_store.getPaymentMethodsList();
         my_profile_store.getSelectedPaymentMethodDetails();
         my_profile_store.setAddPaymentMethodErrorMessage('');
         my_profile_store.setIsCancelAddPaymentMethodModalOpen(false);
 
-        return () => {
-            my_profile_store.setSelectedPaymentMethod('');
```

Comment:
why did we remove this?

Reply:
Since we are now using saving the form state using `ModalForm`, this cleanup function will override the saved state